discovery-infra/test_infra/tools/assets.py

```diff
@@ -44,6 +50,110 @@ def release_all(self) -> None:
+        self._allocated_ips_objects.clear()
+
+        self._fill_virsh_allocated_ips_objects_and_allocated_bridges()
+
+        self._override_ip_networks_values_if_not_free(asset)
+        self._override_network_bridges_values_if_not_free(asset)
+
+        logging.info("Using network asset: %s", asset)
+        return asset
+
+    @staticmethod
+    def _verify_asset_fields(asset: Munch):
+        for field in consts.REQUIRED_ASSET_FIELDS:
+            assert hasattr(asset, field), f"missing field {field}"
+
+    def _fill_virsh_allocated_ips_objects_and_allocated_bridges(self):
+        with LibvirtController.connection_context() as conn:
+            for net in conn.listAllNetworks():
+                for lease in net.DHCPLeases():
+                    net_bridge = lease.get('iface')
+                    if net_bridge:
+                        self._add_allocated_net_bridge(net_bridge)
+
+                    ipaddr = lease.get('ipaddr')
+                    if ipaddr:
+                        self._add_allocated_ip(IPAddress(ipaddr))
+
+    def _override_ip_networks_values_if_not_free(self, asset: Munch):
+        self._find_current_allocated_host_ips()
+
+        logging.info("IPs in use: %s", self._allocated_ips_objects)
+
+        for ip_network_field in consts.IP_NETWORK_ASSET_FIELDS:
+            ip_network = IPNetwork(getattr(asset, ip_network_field))
+            self._set_next_available_ip_network(ip_network)
+            self._add_allocated_ip(ip_network)
+            setattr(asset, ip_network_field, str(ip_network))
+
+    def _find_current_allocated_host_ips(self):
+        cmd_output = sp.check_output(["hostname", "--all-ip-addresses"])
+        for ip in cmd_output.split():
+            self._add_allocated_ip(IPAddress(ip.decode()))
+
+    def _set_next_available_ip_network(self, ip_network: IPNetwork):
+        while self._is_ip_network_allocated(ip_network):
+            ip_network += 1
+
+    def _is_ip_network_allocated(self, ip_network: IPNetwork) -> bool:
+        for ip_addr in self._allocated_ips_objects:
+            if ip_addr in ip_network:
+                return True
+
+        return False
+
+    def _add_allocated_ip(self, ip: Union[IPNetwork, IPRange, IPAddress]):
+        self._allocated_ips_objects.append(ip)
+
+    def _override_network_bridges_values_if_not_free(self, asset: Munch):
+        logging.info("Bridges in use: %s", self._allocated_bridges)
+
+        if self._is_net_bridge_allocated(asset.libvirt_network_if):
+            asset.libvirt_network_if = self._get_next_available_net_bridge()
+            self._add_allocated_net_bridge(asset.libvirt_network_if)
+
+        if self._is_net_bridge_allocated(asset.libvirt_secondary_network_if):
+            net_bridge = self._get_next_available_net_bridge(is_secondary=True)
+            asset.libvirt_secondary_network_if = net_bridge
+
+    def _get_next_available_net_bridge(
+        self,
+        *,
+        is_secondary: bool = False
+    ) -> str:
```

Comment:
partially you set return value from the function and partially not. Please decide which syntax you want to stick to it

Reply:
I actually not set return values only when it is None. 
Will add it explicitly.